src/number_theory/zeta_function.lean

```diff
@@ -0,0 +1,373 @@
+    rw [pi.add_apply, zeta_kernel₂, pi.add_apply, indicator_of_mem h],
+    ring },
+  refine ((is_O_zero_zeta_kernel₂_rpow _).add _).congr' this.symm (eventually_eq.refl _ _),
+  simp_rw sub_div,
+  apply is_O.sub,
+  { apply is_O.of_norm_left,
+    simp_rw [norm_div, norm_one, div_eq_mul_inv, one_mul, mul_comm _ (‖(2 : ℂ)‖)⁻¹],
+    refine ((is_O_refl _ _).congr' (eventually_eq.refl _ _)
+      (eventually_eq_of_mem self_mem_nhds_within (λ x hx, _))).const_mul_left _,
+    rw [is_R_or_C.norm_of_real, abs_of_nonneg (sqrt_nonneg _)],
+    simp_rw [sqrt_eq_rpow, rpow_neg (le_of_lt hx), one_div] },
+  { refine is_O_iff.mpr ⟨‖(1 / 2 : ℂ)‖, _⟩,
+    refine eventually_of_mem (Ioc_mem_nhds_within_Ioi $ left_mem_Ico.mpr zero_lt_one) (λ t ht, _),
+    refine le_mul_of_one_le_right (norm_nonneg _) _,
+    rw [norm_of_nonneg (rpow_nonneg_of_nonneg ht.1.le _), rpow_neg ht.1.le],
+    exact one_le_inv (rpow_pos_of_pos ht.1 _) (rpow_le_one ht.1.le ht.2 one_half_pos.le) }
+end
+
+/-!
+## Differentiability of the completed zeta function
+-/
+
+/-- The Mellin transform of the first zeta kernel is holomorphic for `1 / 2 < re s`. -/
+lemma differentiable_at_mellin_zeta_kernel₁ {s : ℂ} (hs : 1 / 2 < s.re) :
+  differentiable_at ℂ (mellin zeta_kernel₁) s :=
+mellin_differentiable_at_of_is_O_rpow_exp pi_pos locally_integrable_zeta_kernel₁
+  is_O_at_top_zeta_kernel₁ is_O_zero_zeta_kernel₁ hs
+
+/-- The Mellin transform of the second zeta kernel is entire. -/
+lemma differentiable_mellin_zeta_kernel₂ : differentiable ℂ (mellin zeta_kernel₂) :=
+λ s, mellin_differentiable_at_of_is_O_rpow_exp pi_pos locally_integrable_zeta_kernel₂
+  is_O_at_top_zeta_kernel₂ (is_O_zero_zeta_kernel₂_rpow _) ((sub_lt_self_iff _).mpr zero_lt_one)
+
+/-- The modified completed Riemann zeta function `Λ(s) + 1 / s - 1 / (s - 1)` is entire. -/
+theorem differentiable_completed_zeta₀ : differentiable ℂ riemann_completed_zeta₀ :=
+differentiable_mellin_zeta_kernel₂.comp (differentiable.div_const differentiable_id 2)
+
+/-- The completed Riemann zeta function `Λ(s)` is differentiable away from `s = 0` and `s = 1`
+(where it has simple poles). -/
+theorem differentiable_at_completed_zeta {s : ℂ} (hs : s ≠ 0) (hs' : s ≠ 1) :
+  differentiable_at ℂ riemann_completed_zeta s :=
+begin
+  refine (differentiable_completed_zeta₀.differentiable_at.sub _).add _,
+  { exact (differentiable.differentiable_at (differentiable_const _)).div differentiable_at_id hs },
+  { refine ((differentiable_const _).differentiable_at).div _ (sub_ne_zero.mpr hs'),
+    exact differentiable_at_id.sub (differentiable_at_const _) },
+end
+
+/-!
+## Relating the Mellin transforms of the two zeta kernels
+-/
+
+lemma mellin_zeta_kernel₂_eq_of_lt_re {s : ℂ} (hs : 1 / 2 < s.re) :
+  mellin zeta_kernel₂ s = mellin zeta_kernel₁ s + 1 / (2 * s) - 1 / (2 * s - 1) :=
+begin
+  -- There is nothing really deep happening in this argument, just a lot of manipulations of
```

Comment:
Could you streamline this by giving a formula for the Mellin transform of a sum or difference (under suitable integrability assumption) and for the Mellin transform of the auxiliary function `indicator (Ioc 0 1) (λ t, (1 - 1 / sqrt t) / 2)`?

Reply:
Working on it 🛠️

Reply:
I now have a more civilised proof of this, but it involves a few additions to Mellin transforms, and to the general integration machinery as prerequisites. I'll make some other PR's for the new stuff + come back to this once they're merged.

Reply:
I've just pushed the civilised version using the general lemmas on Mellin transforms introduced at #19110.